sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -222,20 +214,37 @@ def transform_messages_to_sendable_if_needed(messages):
         except AttributeError:
             return messages
 
+
+def _single_message_from_dict(message, message_type):
+    # type: (Union[ServiceBusMessage, Mapping[str, Any]], Type[ServiceBusMessage]) -> ServiceBusMessage
+    if isinstance(message, message_type):
+        return message
+    try:
+        return message_type(**cast(Mapping[str, Any], message))
+    except TypeError:
+        raise TypeError(
+            "Only ServiceBusMessage instances or Mappings are supported. "
+            "Received instead: {}".format(
+                message.__class__.__name__
+            )
+        )
+
+
 def create_messages_from_dicts_if_needed(messages, message_type):
-    # type: (DictMessageType, type) -> DictMessageReturnType
+    # type: (MessagesType, Type[ServiceBusMessage]) -> Union[ServiceBusMessage, List[ServiceBusMessage]]
     """
     This method is used to convert dict representations
     of messages to a list of ServiceBusMessage objects or ServiceBusBatchMessage.
```

Comment:
since `ServiceBusMessageBatch` is not of `MessagesType`, should it be removed from this description?
```suggestion
    of messages to a list of ServiceBusMessage objects.
```

Reply:
Thanks - fixed!